pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test: TestRepeatableReadLogic_grant_in_txn failed

pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test.TestRepeatableReadLogic_grant_in_txn [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20785365?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20785365?buildTab=artifacts#/) on release-25.4 @ [de304620a173c9b6187d1fc363d6ed7b9152f418](https://github.com/cockroachdb/cockroach/commits/de304620a173c9b6187d1fc363d6ed7b9152f418):


```
=== RUN   TestRepeatableReadLogic_grant_in_txn
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/20ed47d97df69a73c822c8f00ec93dea/logTestRepeatableReadLogic_grant_in_txn1601555526
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[02:41:16] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18677/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn: 49 statements
[02:41:17] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18677/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn with config local-repeatable-read: 69 tests, 0 failures
    logic.go:4486: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18677/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:231: error while processing
    logic.go:4486: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18677/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/local-repeatable-read/local-repeatable-read_test_/local-repeatable-read_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/grant_in_txn:231: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: cannot publish new versions for descriptors: [{role_members 23 37}], old versions still in use
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/20ed47d97df69a73c822c8f00ec93dea/logTestRepeatableReadLogic_grant_in_txn1601555526
--- FAIL: TestRepeatableReadLogic_grant_in_txn (3.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRepeatableReadLogic_grant_in_txn.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57278

--- a/pkg/sql/logictest/testdata/logic_test/grant_in_txn
+++ b/pkg/sql/logictest/testdata/logic_test/grant_in_txn
@@ -191,6 +191,14 @@ GRANT role_bar TO role_foo WITH ADMIN OPTION;
 statement ok
 GRANT role_foo TO testuser WITH ADMIN OPTION;
 
+# GRANTS only wait for the new version to be visible, but future schema changes
+# can retry errors if the old version is in use, due to the two version invariant.
+retry
+query I
+SELECT count(*) FROM system.lease WHERE desc_id = 'system.role_members'::REGCLASS::OID GROUP BY desc_id, version;
+----
+1
+
 # switch to testuser
 
 user testuser

